File: poc-kronos-frontend/src/model/pessoa/Filiacao.js
class Filiacao {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.dataFiliacao = dados ? dados.dataFiliacao : null;
    this.filiacaoAtual = dados ? dados.filiacaoAtual : null;
    this.idPessoaPais = dados ? dados.idPessoaPais : null;
    this.nomePessoaPais = dados ? dados.nomePessoaPais : null;
    this.idPessoaFilho = dados ? dados.idPessoaFilho : null;
    this.nomePessoaFilho = dados ? dados.nomePessoaFilho : null;
    this.idTipoFiliacao = dados ? dados.idTipoFiliacao : null;
    this.nomeTipoFiliacao = dados ? dados.nomeTipoFiliacao : null;
  }

  static validations(required) {
    return {
      dataFiliacao: { required },
      idPessoaPais: { required },
      idPessoaFilho: { required },
      idTipoFiliacao: { required }
    };
  }

  static validationMessages() {
    return {
      dataFiliacao: {
        required: "Data Filiação é obrigatória"
      },
      idPessoaPais: {
        required: "Pessoa é obrigatória"
      },
      idPessoaFilho: {
        required: "Filho é obrigatório"
      },
      idTipoFiliacao: {
        required: "Tipo Filiação é obrigatória"
      }
    };
  }
}
export default Filiacao;

File: src/main/java/br/com/kronos/backend/aplicacao/pessoa/Filiacao.java
package br.com.kronos.backend.aplicacao.pessoa;

import static java.util.Objects.requireNonNull;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Filiacao {
	private int id;
    private LocalDate dataFiliacao;
    private boolean filiacaoAtual;
    private Long idPessoaPais;
    private Long idPessoaFilho;
    private int idTipoFiliacao;

	@Builder
	public Filiacao(int id, LocalDate dataFiliacao, boolean filiacaoAtual, Long idPessoaPais, Long idPessoaFilho, int idTipoFiliacao) {

		super();
        this.id = id;
        this.dataFiliacao = requireNonNull(dataFiliacao, "dataFiliacao é obrigatório");
        this.filiacaoAtual = requireNonNull(filiacaoAtual, "filiacaoAtual é obrigatório");
        this.idPessoaPais = requireNonNull(idPessoaPais, "idPessoaPais é obrigatório"); 
        this.idPessoaFilho = requireNonNull(idPessoaFilho, "idPessoaFilho é obrigatório");
        this.idTipoFiliacao = requireNonNull(idTipoFiliacao, "idTipoFiliacao é obrigatório");
       
	}
}